### PR TITLE
Add "legacy" parameter required by Xcode 16

### LINKED ISSUE #96: xcresult parsing fails with Xcode 16
### New Issue Checklist

- [x] Updated the plugin to the latest version 
- [x] I searched for existing [GitHub issues](https://github.com/insideapp-oss/sonar-apple/issues)

### Issue Description

When running analysis with Xcode 16, xcresult parsing fails, and as a result some measures are missing.

##### Command executed

Simple sonar-scanner execution

##### Complete output when running sonar-scanner, including the stack trace and command used

```
org.buildobjects.process.ExternalProcessFailureException: External process `xcrun` terminated with unexpected exit status 64 after 50ms:
  $ xcrun xcresulttool get --format json --path /Users/ggrousset/Projects/studi/studi-lms-ios/build/result.xcresult
  STDERR: Error: This command is deprecated and will be removed in a future release, --legacy flag is required to use it.
  STDERR: Usage: xcresulttool get object [--legacy] --path <path> [--id <id>] [--version <version>] [--format <format>]
  STDERR:   See 'xcresulttool get object --help' for more information.
```

### Environment

macOS 14.5 with Xcode 16.0

### PR DESCRIPTION
**Pull request by gaelfoppolo**
# Summary 

This PR fixes #96 

# DoD

- [x] Implementation
- [x] Unit tests

### CODE REVIEW

File: sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public XcodeResultReadObjectRunner() {
     protected String[] arguments(File resultBundle, Reference reference) {
         return new String[]{
                 "xcresulttool", "get",
+                "--legacy",
```
Comment (zippy1978) [inline sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java:35]: will it still work with Xcode < 16 ?

Comment (gaelfoppolo) [reply sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java:35]: Indeed, I check and it doesn't work.
I checked how other tools managed to handle both.

- fastlane: result of `xcrun xcresulttool version` and expect a minimum version (23021).
https://github.com/fastlane/fastlane/pull/22147/files#diff-3c9438dd39c9680b4944d642b8eac74d953cd446703bb1901842d7fa03f8c248R217

- xcresulttool: result of `xcodebuild -version` and expect a minimum version (16).
https://github.com/gemmakbarlow/xcresulttool/pull/1/files#diff-abe2818e9c29aadf84401cb7ef8286c7cbce9253eba09f6fa52e22ca4e57b17eR82

Which solution seems to be the best for us?

Comment (zippy1978) [reply sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java:35]: `xcrun xcresulttool version`seems the easier to parse :)

Comment (omartio) [reply sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java:35]: @gaelfoppolo Hello! 
Do you have any updates on this issue?
I'm looking forward to using a fixed version.

Comment (gaelfoppolo) [reply sonar-apple-plugin/src/main/java/fr/insideapp/sonarqube/apple/xcode/runner/XcodeResultReadObjectRunner.java]: I implemented a checker based on what fastlane does. Tested with Xcode 15 and 16, both are now working.

**Pull request merged**